dts/arm/atmel/samd2x.dtsi

```diff
@@ -72,7 +72,7 @@
 				compatible = "soc-nv-flash";
 				label = "FLASH_0";
 				reg = <0 0x40000>;
-				write-block-size = <64>;
+				write-block-size = <4>;
```

Comment:
I am curious where to find this in datasheet?

Reply:
[SAMD20 Datasheet](https://ww1.microchip.com/downloads/en/DeviceDoc/SAM_D20_%20Family_Datasheet_DS60001504C.pdf) - 20.6.4.2 NVM Write (page 262). Writes addresses are aligned to a maximum of 32-bits. Hardware write address is 16-bits (per `ADDR` register).

Reply:
This is wrong. SAM 2x flash page size is 64 bytes. Snip from SAM D21 Family Data Sheet:

"""
The NVM is organized into rows, where each row contains four pages, as shown in the NVM Row Organization
figure. The NVM has a row-erase granularity, while the write granularity is by page. In other words, a single row erase
will erase all four pages in the row, while four write operations are used to write the complete row.
"""

However, when writing a page, one has to first write to a page buffer 16 or 32-bits at a time.

"""
Data to be written to the NVM block are first written to and stored in an internal buffer called the page buffer. The
page buffer contains the same number of bytes as an NVM page. Writes to the page buffer must be 16 or 32 bits. 8-
bit writes to the page buffer are not allowed and will cause a system exception.
"""

The `flash_sam0` and `flash_sam` drivers handle this in `flash_sam0_write_page` and `flash_sam_write_page`

Reply:
You can write less than a full page to SAMD20/21 devices. From 20.6.4.2.2: 
>> Note that partially written pages must be written with a manual write.

The manual write process is outlined in [SAMD20 datasheet](https://ww1.microchip.com/downloads/en/DeviceDoc/SAM_D20_%20Family_Datasheet_DS60001504C.pdf) - 20.6.4.2.1. 

`flash_sam0_write_page` assumes you are writing a full 64 byte page, although the hardware allows you to write partial pages.

Reply:
Writes must be aligned to a page boundary. If one wants to do a write a 32 bit value in the middle of a page by writing a page partially, preceding bytes must be 0xff. The `flash_sam0` driver do not handle this, as far as I can see.

This pull request breaks the `flash_sam0`driver. You the `flash_sam0_write_page` assumes a 64 byte page, but `samd2x.dtsi` states that it is allowed to write 4 bytes.

Reply:
I have a PR that will address this for the `flash_sam0` driver. In the interm `CONFIG_SOC_FLASH_SAM0_EMULATE_BYTE_PAGES` should allow it to write with 1 byte granularity at reduced performance cost.